### ISSUE TITLE
connect bitcoin node using cookie file

### ISSUE THREAD

**Issue by louisinger** [ark-network/ark#296]
`BITCOIND_RPC_USER`, `BITCOIND_RPC_PASS` is not a safe way to connect bitcoind. For production, we should handle `.cookie` file.

**Comment by miyo-fuji** [comment]
Yes, but imho we should have both options. For example Voltage only support RPC users

**Comment by tiero** [comment]
Yes, but imho we should have both options. For example Voltage only support RPC users

**Comment by Randy808** [comment]
Can I get assigned this issue?

**Comment by Randy808** [comment]
Actually I'm confused by what this issue is asking for. Is the assumption that we'd be running arkd from the same host where bitcoind is running?

**Comment by tiero** [comment]
> what this issue is asking for

Now the logic assumes that either you pass a Neturino peer or you pass JSONRPC credentials (along with ZMQ)
https://github.com/ark-network/ark/blob/master/server/internal/config/config.go#L501-L510


> Is the assumption that we'd be running arkd from the same host where bitcoind

not at all, if you specify `-rpcbind` correctly on your `bitcoind`  it should let you connect without `arkd` having extra check.  Not currently aware if Bitcoin Core via cookie force you to be on the same host? Then it should have a descriptive error message

**Comment by louisinger** [comment]
We need a new `BITCOIN_COOKIE_FILEPATH` env variable letting us specify the .cookie location. 

As far as I know, .cookie can be used even if not on the same host (ofc, it needs the .cookie file to be moved on the arkd host)

**Comment by Randy808** [comment]
The cookie file is read (or [generated](https://github.com/bitcoin/bitcoin/blob/770d39a37652d40885533fecce37e9f71cc0d051/src/httprpc.cpp#L308)) by bitcoind [during startup](https://github.com/bitcoin/bitcoin/blob/770d39a37652d40885533fecce37e9f71cc0d051/src/httprpc.cpp#L362) and [used by the locally-run bitcoin-cli](https://github.com/bitcoin/bitcoin/blob/770d39a37652d40885533fecce37e9f71cc0d051/src/bitcoin-cli.cpp#L855) to connect to the bitcoind rpc server. It's an [implementation detail](https://github.com/bitcoin/bitcoin/blob/770d39a37652d40885533fecce37e9f71cc0d051/doc/JSON-RPC-interface.md?plain=1#L147) for making authentication easy when bitcoin-cli (or another bitcoin rpc client) is used on the same host* since it can scan the datadir bitcoind was configured with. For remote applications, [`rpcauth` is what will be replacing rpcuser and rpcpass](https://github.com/bitcoin/bitcoin/blob/770d39a37652d40885533fecce37e9f71cc0d051/src/httprpc.cpp#L317)

\* Or from a host with a shared volume

**Comment by louisinger** [comment]
You're right, ideally we need to support both rpcauth and cookie file
